packages/front-end/pages/fact-tables/index.tsx

```diff
@@ -254,26 +256,28 @@ export default function FactTablesPage() {
             >
               Auto-generate Fact Tables
             </Button>
-          </Tooltip>
-          <Tooltip
-            body={
-              !canCreate
-                ? `You don't have permission to create fact tables ${
-                    project ? "in this project" : ""
-                  }`
-                : ""
-            }
-          >
-            <Button
-              onClick={() => {
-                if (!canCreate) return;
-                setCreateFactOpen(true);
-              }}
-              disabled={!canCreate}
+          )}
+          {hasDatasource ? (
+            <Tooltip
+              body={
+                !canCreate
+                  ? `You don't have permission to create fact tables ${
+                      project ? "in this project" : ""
+                    }`
+                  : ""
+              }
             >
-              Add Fact Table
-            </Button>
-          </Tooltip>
+              <Button
+                onClick={() => {
+                  if (!canCreate) return;
+                  setCreateFactOpen(true);
+                }}
+                disabled={!canCreate}
```

Comment:
nit: Should we just have this disabled instead of invisible? So that way the "connect a datasource first" message and CTA are more just added to this page, but it always looks like it will in the future, minus this button being disabled? That's kind of what I expect from apps, but might not be worth it.

Reply:
People tend not to read, so I think having a single CTA is better and will be less confusing to users.